src/libxively/sft/xi_sft_logic.c

```diff
@@ -131,6 +144,89 @@ xi_sft_send_file_status( const xi_sft_context_t* context,
     ( *context->fn_send_message )( context->send_message_user_data, message_file_status );
 }
 
+xi_state_t xi_sft_select_next_resource_to_download( xi_sft_context_t* context )
+{
+    int32_t selected_index       = -1;
+    context->update_current_file = NULL;
+    uint16_t i                   = 0;
+    for ( ; i < context->update_message_fua->file_update_available.list_len; ++i )
+    {
+        if ( 0 <= context->updateable_files_download_order[i] )
+        {
+            selected_index = context->updateable_files_download_order[i];
+            context->updateable_files_download_order[i] = -1;
+            break;
+        }
+    }
+
+    if ( selected_index >= context->update_message_fua->file_update_available.list_len )
```

Comment:
this is a check if the selected index is out of bounds, this check is already done in the other function, I will write comment there too. I think either of these two is enough.

Reply:
Ok, removing this one.